### DISCUSSION TITLE
reactpy with flask

### DISCUSSION THREAD

**Discussion by yickysan** [idom-team/idom#1141] [Question] [answered]
Hello beautiful community.  I have a question regarding reactpy with flask framework. I normally use flask to deploy ml apps. Is it possible to still be able to define my routes using flask and have it do server side data processing before sending information to reactpy in the frontend?

**Comment by Archmonger** [comment, marked as answer]
Yes, you can do this by embedding ReactPy into your Flask HTML templates using our script
```html
<script type="module">
  import { mount } from "https://esm.sh/@reactpy/client";
  const client = new SimpleReactPyClient({
    serverLocation: {
      url: document.location.origin,
      route: document.location.pathname,
      query: document.location.search,
    },
  });
  mount(element, client);
</script>
```

Please note that embedding ReactPy is currently easier to do with the Django backend, by using our [template tag](https://reactive-python.github.io/reactpy-django/3.5.1/reference/template-tag/#component).
